File: backend/Snagged.backend/Snagged.Domain/Entities/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Snagged.Domain.Entities
{
    public class Order
    {
        public int Id { get; set; }
        public int BuyerId { get; set; }
        public DateTime OrderDate { get; set; } = DateTime.Now;
        public string Status { get; set; }
        public int? PaymentId { get; set; }
        public User Buyer { get; set; }
        public Payment Payment { get; set; }
        public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
    }
}

File: frontend/src/app/shared/models/order.ts
import { User } from './user';
import { Payment } from './payment';
import { OrderItem } from './order-item';

export interface Order {
  id: number;
  buyerId: number;
  orderDate: Date | string;
  status: string;
  paymentId?: number;
  buyer?: User;
  payment?: Payment;
  orderItems: OrderItem[];
  stripePaymentIntentId?: string;
}
